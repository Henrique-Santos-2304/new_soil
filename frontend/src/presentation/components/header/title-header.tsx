import appFonts from '@/shared/styles/theme/custom/my-fonts'
import { Box, Typography } from '@mui/material'

interface HeaderProps {
  title: string
  subTitle?: string
  contentSubTitle?: string
}

const TitleHeader: React.FC<HeaderProps> = ({ title, subTitle, contentSubTitle }) => {
  return (
    <Box
      id="titulo e subtitulo da página"
      display="flex"
      flex={1}
      flexDirection="column"
      alignItems="center"
      justifyContent="space-evenly"
    >
      <Typography
        aria-label="Título da página"
        variant="h1"
        fontSize="2.2rem"
        color="primary"
        fontFamily={appFonts.poppins}
        fontWeight="600"
      >
        {title}
      </Typography>
      {subTitle && (
        <Box
          aria-label="Sub-Título da página"
          mt={2}
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Typography variant="h4" fontSize="1.1rem" color="primary" fontWeight="700">
            {subTitle}
          </Typography>
          {contentSubTitle && (
            <Typography variant="h4" ml={1} fontSize="1rem" color="primary" fontWeight="500">
              {contentSubTitle}
            </Typography>
          )}
        </Box>
      )}
    </Box>
  )
}

export { TitleHeader }
