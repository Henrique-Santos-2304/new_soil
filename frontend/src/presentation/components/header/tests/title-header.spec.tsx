import { screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { renderThemeMock } from '@/tests/index'
import { TitleHeader } from '../title-header'

describe('Title Header Component', () => {
  it('should render component and write snapshot', () => {
    const { container } = renderThemeMock(<TitleHeader title="Olá" />)

    expect(container).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
  it('should expected header to have heading with title received', () => {
    renderThemeMock(<TitleHeader title="Olá" />)

    const heading = screen.getByRole('heading', {
      level: 1
    })

    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('Olá')
  })

  it('should expected header to have heading with subtitle received', () => {
    renderThemeMock(<TitleHeader title="Olá" subTitle="send-subtitle" />)

    const heading = screen.getByRole('heading', {
      level: 4
    })

    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('send-subtitle')
  })

  it('should expected header to have heading with content-subtitle received', () => {
    renderThemeMock(<TitleHeader title="Olá" subTitle="send-content-subtitle" />)

    const heading = screen.getByRole('heading', {
      level: 4
    })

    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('send-content-subtitle')
  })
})
