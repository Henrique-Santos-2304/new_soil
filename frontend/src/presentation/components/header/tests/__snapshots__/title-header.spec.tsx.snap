// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Header Component should render component and write snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-imz5j4"
  >
    <div
      class="MuiBox-root css-49rcs4"
      id="titulo e subtitulo da página"
    >
      <h1
        aria-label="Título da página"
        class="MuiTypography-root MuiTypography-h1 css-7inatm-MuiTypography-root"
      >
        Olá
      </h1>
    </div>
  </div>
</div>
`;
